rs/replicated_state/src/page_map/page_allocator/mmap.rs

```diff
@@ -563,6 +568,30 @@ impl MmapBasedPageAllocatorCore {
                 mmap_size, self.file_descriptor, mmap_file_offset, err,
             )
         }) as *mut u8;
+
+        // Do madvise transparent huge page performance optimization only on Linux.
+        #[cfg(target_os = "linux")]
+        {
+            // Huge pages are 2MiB on x86_64. We only use huge pages for
+            // memory allocations that are at least MIN_NUM_HUGE_PAGES_FOR_OPTIMIZATION huge pages (i.e., 64 MiB).
+            if mmap_size >= MIN_NUM_HUGE_PAGES_FOR_OPTIMIZATION.get() as usize * HUGE_PAGE_SIZE {
+                unsafe {
+                    madvise(
+                        mmap_ptr as *mut c_void,
+                        mmap_size,
+                        MmapAdvise::MADV_HUGEPAGE,
+                    )
+                }.unwrap_or_else(|err| {
+                    // We don't need to panic, madvise failing is not a problem, 
+                    // it will only mean that we are not using huge pages.
+                    println!(
```

Comment:
Do we have a logger that we could use instead of println?

Reply:
There's no logger handy; these calls come all the way from `PageMap` which doesn't have a logger attached. Also `PageMap` objects live in both sandboxes and replica, which makes it messy to add logging. I'd propose to defer this to some time in the future.